tests/test_cli.py

```diff
@@ -150,6 +150,39 @@ class Foo(SQLModel, table=True):
     )
 
 
+def test_cli_engine_arg(db_path: Path, tmp_path: Path) -> None:
+    output_path = tmp_path / "outfile"
+    subprocess.run(
+        [
+            "sqlacodegen",
+            f"sqlite:///{db_path}",
+            "--generator",
+            "tables",
+            "--engine-arg",
+            'connect_args={"timeout": 10}',
```

Comment:
How does this test if the correct argument was passed to `create_engine()`? I think it should be an argument that actually makes a difference in the output such that we can check.

Reply:
Possible arguments for sqlite: https://github.com/zzzeek/sqlalchemy/blob/main/lib/sqlalchemy/dialects/sqlite/pysqlite.py#L610-L617

Since these are connection options, it shouild not really affect how code is generated from DDL. It is the same for Oracle, `thick_mode` only specifies how connections are made. 

FYI if you specify an invalid argument not recognised by the database driver, `create_engine(...)` fails - would you like to add some tests for verifying such failures (such as calling `sqlite` driver with invalid arguments?)

Reply:
Some test that checks that the engine args aren't just discarded would be nice.

Reply:
I'm ready to merge once this is done.

Reply:
Added a test to check that even invalid `--engine-arg` is passed and the driver fails for that